src/main/webapp/js/util.js

```diff
@@ -76,6 +76,107 @@ function createDealCard(deal) {
+  const dealCardCol = document.createElement('div');
+  dealCardCol.className = 'col-md-' + numCol;
+
+  const dealCard = document.createElement('div');
+  dealCard.classList.add('deal-card', 'card', 'h-100');
+  dealCard.id = 'deal-card-' + i;
+
+  const dealImage = document.createElement('img');
+  dealImage.className = 'card-img-top home-deal-img';
+
+  const dealBody = document.createElement('div');
+  dealBody.classList.add('card-body', 'd-flex', 'flex-column');
+
+  const dealTime = document.createElement('div');
+  dealTime.classList.add('card-text');
+  dealTime.innerText = deal.timestamp;
+
+  const dealVotes = document.createElement('div');
+  dealVotes.classList.add('d-flex', 'justify-content-end');
+  dealVotes.style = 'display: none;';
+
+  const dealUpVotesBtn = document.createElement('button');
+  dealUpVotesBtn.classList.add('btn', 'upvote-btn');
+
+  const upBtnAngle = document.createElement('span');
+  upBtnAngle.classList.add('fas', 'fa-angle-up');
+
+  dealUpVotesBtn.appendChild(upBtnAngle);
+
+  const dealVotesValue = document.createElement('span');
+  dealVotesValue.classList.add('my-auto', 'votes-num');
+  dealVotesValue.innerText = deal.votes;
+
+  const dealDownVotesBtn = document.createElement('button');
+  dealDownVotesBtn.classList.add('btn', 'downvote-btn');
+
+  const downBtnAngle = document.createElement('span');
+  downBtnAngle.classList.add('fas', 'fa-angle-down');
+
+  dealDownVotesBtn.appendChild(downBtnAngle);
+
+  dealVotes.appendChild(dealUpVotesBtn);
+  dealVotes.appendChild(dealVotesValue);
+  dealVotes.appendChild(dealDownVotesBtn);
+
+  const dealName = document.createElement('h5');
+  dealName.className = 'card-title';
+  dealName.innerText = deal.description;
+
+  const dealRestaurant = document.createElement('div');
+  dealRestaurant.innerText = 'Restaurant: ' + deal.restaurant.name;
+
+  const dealRestaurantLink = document.createElement('a');
+  dealRestaurantLink.href = '/restaurants/' + deal.restaurant.id;
+
+  dealRestaurant.appendChild(dealRestaurantLink);
+
+  const dealPoster = document.createElement('div');
+  dealPoster.innerText = 'Poster: ' + deal.poster.username;
+
+  const dealPosterLink = document.createElement('a');
+  dealPosterLink.href = '/users/' + deal.poster.id;
+
+  dealPoster.appendChild(dealPosterLink);
+
+  const dealTags = document.createElement('div');
+
+  for (let i = 0; i < deal.tags.length; i++) {
+    dealTags.innerText += '#' + deal.tags[i].name + ', ';
+  };
+
+  const dealSeeMore = document.createElement('a');
+  dealSeeMore.href = '/deals/' + deal.id;
+  dealSeeMore.innerText = 'See More';
+  dealSeeMore.classList.add('btn', 'btn-primary', 'align-self-end',
+      'mt-auto', 'float-right');
+
+  dealBody.appendChild(dealTime);
+  dealBody.appendChild(dealVotes);
+  dealBody.appendChild(dealName);
+  dealBody.appendChild(dealRestaurant);
+  dealBody.appendChild(dealPoster);
+  dealBody.appendChild(dealTags);
+  dealBody.appendChild(dealSeeMore);
```

Comment:
can you rewrite the function as something like
```
dealBody.innerHTML = `
    <div class="card-text">${deal.timestamp}</div>
    <div class="d-flex justify-content-end">...</div>
    ...
`
```
It is a bit hard to read 100 lines of `document.createElement` and `.appendChild`